Samples/ios/SampleAppiOS-Swift/SampleAppiOS-Swift/SampleCalendarUtil.swift

```diff
@@ -0,0 +1,169 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+import Foundation
+
+typealias CalendarCompletion = ([Date: [SampleCalendarEvent]]?, Error?) -> Void
+
+class SampleCalendarUtil  {
+    
+    // Constants
+    fileprivate let kLastEventsCheckKey = "last_events_check"
+    fileprivate let kEventsKey = "events"
+    
+    // Singleton instance
+    static let shared = SampleCalendarUtil()
+    
+    private init() {
+        if let storedEvents = UserDefaults.standard.object(forKey: kEventsKey) as? [[String: Any]] {
+            self.cachedEvents = processEvents(withEvents: storedEvents)
+        }
+        else {
+            self.cachedEvents = [Date: [SampleCalendarEvent]]()
+        }
+    }
+    
+    /*
+     Returns cached events (if any) for the current user
+     */
+    var cachedEvents: [Date: [SampleCalendarEvent]]!
+    
+    /*
+     Clears any cached events for the current user
+     */
+    func clearCache() {
+        cachedEvents.removeAll()
```

Comment:
Should also remove value for kLastEventsCheckKey here, so when another user logs in, we check his events.